src/silx/math/fit/functions.pyx

```diff
@@ -472,6 +473,62 @@ def sum_splitpvoigt(x, *params):
     return numpy.asarray(y_c).reshape(x.shape)
 
 
+def sum_splitpvoigt2(x, *params):
+    """Return a sum of split pseudo-Voigt functions, defined by *(height,
+    centroid, fwhm1, fwhm2, eta1, eta2)*.
+
+    The pseudo-Voigt profile ``PV(x)`` is an approximation of the Voigt
+    profile using a linear combination of a Gaussian curve ``G(x)`` and a
+    Lorentzian curve ``L(x)`` instead of their convolution.
+
+        - *height* is the peak amplitude for G(x) and L(x)
+        - *centroid* is the peak x-coordinate for both functions
+        - *fwhm1* is the full-width at half maximum of both functions
+          when ``x < centroid``
+        - *fwhm2* is the full-width at half maximum of both functions
+          when ``x > centroid``
+        - *eta1* is the Lorentzian fraction when ``x < centroid``
+        - *eta2* is the Lorentzian fraction when ``x > centroid``
+
+    :param x: Independent variable where the gaussians are calculated
+    :type x: numpy.ndarray
+    :param params: Array of pseudo-Voigt parameters (length must be a multiple
+        of 6):
+        *(height1, centroid1, fwhm11, fwhm21, eta11, eta21,...)*
+    :return: Array of sum of split pseudo-Voigt functions at each ``x``
+        coordinate
+    """
+    cdef:
+        double[::1] x_c
+        double[::1] params_c
+        double[::1] y_c
+
+    if not len(params):
```

Comment:
According to the exception, this would be better suited:
```suggestion
    if len(params) < 6:
```

Reply:
Applied that change to all functions, also modifying the error message accordingly

```
-    if not len(params):
-        raise IndexError("No parameters specified. " +
-                         "At least 6 parameters are required.")
+    if len(params) < 6:
+        raise IndexError("At least 6 parameters are required.")
```